src/main/java/com/sugarcrm/candybean/automation/webdriver/WebDriverElement.java

```diff
@@ -353,20 +354,47 @@ public void scroll(Location loc) throws CandybeanException {
 	}
 
 	/**
-	 * Selects all text in this element if it is a textfield.  Note that this method does not
-	 * currently support textareas or any type of element apart from a textfield.  It clicks on the
-	 * element, so links will be activated, radio buttons will be selected, drop boxes will be
-	 * opened, etc.  Use of this method on any non-textfield element is not currently supported.
+	 * Selects all text in and focuses on a Javascript HTMLInputElement.
+	 *
+	 * @throws CandybeanException If used on a non INPUT element
 	 */
-	public void selectAll() {
-		we.click();
-		we.sendKeys(Keys.END,Keys.chord(Keys.SHIFT,Keys.HOME));
+	public void selectAll() throws CandybeanException {
+		String tag = we.getTagName();
+		if (tag.equals("input")) {
+			executeJavascript("arguments[0].select()");
+		} else throw new CandybeanException("Cannot select text of a non-" +

```

Comment:
I'm not a big fan of this format. Let's keep things consistent and put curly braces around the `throw new CandybeanException`